src/ImageSharp/Metadata/Profiles/Exif/ExifReader.cs

```diff
@@ -189,10 +189,16 @@ protected void ReadSubIfd(List<IExifValue> values)
     {
         if (this.subIfds is not null)
         {
-            foreach (ulong subIfdOffset in this.subIfds)
+            do
             {
-                this.ReadValues(values, (uint)subIfdOffset);
+                ulong[] buf = [.. this.subIfds];
```

Comment:
Can this allocation be avoided?
Either by stack-alloc or renting the array from the array-buffer.

Reply:
reverted in 40b7be804173c56ba8eb5ea2f3b3c305caea3df2 because a build failure,
I haven't found a beautiful solution yet

this.subIfds.lenght is almost always a small number: 1,2, <5

Reply:
You can stackalloc an `ulong[128]` working buffer outside the loop, then in the loop slice it down if `sz <= 128` or allocate an array otherwise.

Reply:
@antonfirsov I forgot to mention that the loop body is executed almost always 1 time, 
or more often it is not even executed at all (subIfds==null),
the task file is the only file where there are 4 loop iterations.

Reply:
> almost always 1 time

Is there any practical limit on the maximum number of `subIfd`-s? `CA2014: Potential stack overflow.` was a valid static analyzer finding if a malicious actor can construct a file with a high number of `subIfd`-s. We need to prepare the code for such edge-cases, while optimizing it for the sane ones.

Reply:
Good point, high number of subIfd-s is a problem. One more problem is cyclic subifds with cross-reference that perform an infinite loop.
I don't think there is any practical point in having a high number of subIfds.
I think 8 subids and a maximum nesting level of 8 are more than enough limits.